hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/JpaTargetType.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.jpa.model;
+
+import org.eclipse.hawkbit.repository.event.remote.TargetTypeDeletedEvent;
+import org.eclipse.hawkbit.repository.event.remote.entity.TargetTypeCreatedEvent;
+import org.eclipse.hawkbit.repository.event.remote.entity.TargetTypeUpdatedEvent;
+import org.eclipse.hawkbit.repository.model.DistributionSetType;
+import org.eclipse.hawkbit.repository.model.Target;
+import org.eclipse.hawkbit.repository.model.TargetType;
+import org.eclipse.hawkbit.repository.model.helper.EventPublisherHolder;
+import org.eclipse.persistence.annotations.CascadeOnDelete;
+import org.eclipse.persistence.descriptors.DescriptorEvent;
+
+import javax.persistence.Column;
+import javax.persistence.ConstraintMode;
+import javax.persistence.Entity;
+import javax.persistence.ForeignKey;
+import javax.persistence.Index;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.constraints.Size;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A target type defines which distribution set types can or have to be
+ * {@link Target}.
+ *
+ */
+@Entity
+@Table(name = "sp_target_type", indexes = {
+        @Index(name = "sp_idx_target_type_prim", columnList = "tenant,id") }, uniqueConstraints = {
+        @UniqueConstraint(columnNames = { "name", "tenant" }, name = "uk_target_type_name")})
+public class JpaTargetType extends AbstractJpaNamedEntity implements TargetType, EventAwareEntity{
+
+    private static final long serialVersionUID = 1L;
+
+    @Column(name = "colour", nullable = true, length = TargetType.COLOUR_MAX_SIZE)
+    @Size(max = TargetType.COLOUR_MAX_SIZE)
+    private String colour;
+
+    @CascadeOnDelete
+    @ManyToMany(targetEntity = JpaDistributionSetType.class)
+    @JoinTable(name = "sp_target_type_ds_type_relation", joinColumns = {
+            @JoinColumn(name = "target_type", nullable = false, updatable = false, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_target_type_relation_target_type"))}, inverseJoinColumns = {
+            @JoinColumn(name = "distribution_set_type", nullable = false, updatable = false, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_target_type_relation_ds_type"))})
+    private Set<DistributionSetType> distributionSetTypes;
+
```

Comment:
`getTargets()` interface method ([see comment above](https://github.com/eclipse/hawkbit/pull/1162/files#r689581201)) needs additional field to access realtionship to Targets.
E.g. (using lazy fetch to only get targets when needed)
```java
@OneToMany(targetEntity = JpaTarget.class, mappedBy = "targetType", fetch = FetchType.LAZY)
private Set<Target> targets;
```